src/invidious/videos.cr

```diff
@@ -782,18 +786,19 @@ struct Video
   end
 end
 
-struct CaptionName
-  include JSON::Serializable
+class Caption
```

Comment:
you should use a `struct` insteed of a `class` for such short-lived data. The former is stored on the stack, where as the latter is stored in momory, which is slower to access.